docs/machine-learning/tutorials/tensorflow-transfer-learning.md

```diff
@@ -0,0 +1,337 @@
+
+Load the data using the `MLContext.Data.ReadFromTextFile` wrapper. It returns a
+<xref:Microsoft.Data.DataView.IDataView> which infers the dataset schema from the `ImageNetData` data model type.
+
+## Build and Train the model
+
+Add the following call to the `BuildAndTrainModel`method as the next line of code in the `Main` method:
+
+[!code-csharp[CallBuildAndTrainModel](../../../samples/machine-learning/tutorials/TransferLearningTF/Program.cs#CallBuildAndTrainModel)]
+
+The `BuildAndTrainModel` method executes the following tasks:
+
+* Loads the data
+* Extracts and transforms the data.
+* Scores the TensorFlow model.
+* Trains the model.
+* Displays model results.
+* Evaluates the model.
+* Saves the model.
+
+Create the `BuildAndTrainModel` method, just after the `Main` method, using the following code:
+
+```csharp
+public static void BuildAndTrainModel(MLContext mlContext, string dataLocation, string imagesFolder, string inputModelLocation, string outputModelLocation)
+{
+
+}
+```
+
+## Extract Features and transform the data
+
+Pre-processing and cleaning data are important tasks that occur before a dataset is used effectively for machine learning. Raw data is often noisy and unreliable, and may be missing values. Using data without these modeling tasks can produce misleading results.
+
+ML.NET's transform pipelines compose a custom `transforms`set that is applied to your data before training or testing. The transforms' primary purpose is data [featurization](../resources/glossary.md#feature-engineering). Machine learning algorithms understand [featurized](../resources/glossary.md#feature) data, so the next step is to transform our textual data into a format that our ML algorithms recognize. That format is a [numeric vector](../resources/glossary.md#numerical-feature-vector).
+
+When the model is trained and evaluated, by default, the values in the **Label** column are considered as correct values to be predicted. As you are using a pretrained model you need to map fields to the new model.. To do that, use the `MLContext.Transforms.Conversion.MapValueToKey`, which is a wrapper for the <xref:Microsoft.ML.ConversionsExtensionsCatalog.ValueToKeyMappingEstimator%2A> transformation class.  The `MapValueToKey` returns an <xref:Microsoft.ML.Data.EstimatorChain%601> that will effectively be a pipeline. Name this `pipeline` as you will then append the trainer to it. Add the next line of code:
+
+[!code-csharp[MapValueToKey1](../../../samples/machine-learning/tutorials/TransferLearningTF/Program.cs#MapValueToKey1)]
+
+Your image processing pipeline uses pre-trained DNN featurizers for feature extraction. The images are loaded in memory as a Bitmap type. Next, you resize the images as the pre-trained model has a defined input image width and height. Finally, The `ImagePixelExtractingEstimator` extracts the pixels from the input images and converts them into a numeric vector. Add these image transforms as the next lines of code:
```

Comment:
Explain why we need to re-size images? also what a DNN means here in this context. This is the first mention of DNN in this tutorial.